libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/RequestHelper.java

```diff
@@ -98,20 +99,25 @@ public static String getIpvSessionId(JourneyRequest request, boolean allowNull)
         return ipvSessionId;
     }
 
-    public static String getFeatureSet(JourneyRequest request) {
-        String featureSet = request.getFeatureSet();
+    public static List<String> getFeatureSet(JourneyRequest request) {
+        List<String> featureSet = request.getFeatureSet();
         LogHelper.attachFeatureSetToLogs(featureSet);
-        return StringUtils.isBlank(featureSet) ? null : featureSet;
+        return (featureSet != null && !featureSet.isEmpty()) ? featureSet : null;
     }
 
-    private static String getFeatureSet(Map<String, String> headers) {
-        return RequestHelper.getHeaderByKey(headers, FEATURE_SET_HEADER);
+    public static List<String> getFeatureSet(APIGatewayProxyRequestEvent event) {
+        List<String> featureSet = getFeatureSet(event.getHeaders());
+        if (featureSet != null && !featureSet.isEmpty()) {
+            LogHelper.attachFeatureSetToLogs(featureSet);
+        }
+        return featureSet;
     }
 
-    public static String getFeatureSet(APIGatewayProxyRequestEvent event) {
-        String featureSet = getFeatureSet(event.getHeaders());
-        LogHelper.attachFeatureSetToLogs(featureSet);
-        return featureSet;
+    public static List<String> getFeatureSet(Map<String, String> headers) {
+        String featureSetHeaderValue = RequestHelper.getHeaderByKey(headers, FEATURE_SET_HEADER);
+        return (featureSetHeaderValue != null)
+                ? Arrays.asList(featureSetHeaderValue.split(","))
+                : null;
     }
```

Comment:
These three methods are inconsistent now - the first _always_ attaches the feature set to logs, the second only does so if nonempty, and the last never does?